src/node/audio_buffer_source.rs

```diff
@@ -506,64 +518,108 @@ impl AudioProcessor for AudioBufferSourceRenderer {
                 && self.render_state.buffer_time < buffer_duration
             {
                 let position = self.render_state.buffer_time * sampling_ratio;
-                self.compute_playback_at_position(position, sample_rate);
+                let playhead = position * sample_rate;
+                let playhead_floored = playhead.floor();
+                let prev_index = playhead_floored as usize; // can't be < 0.
+                let k = (playhead - playhead_floored) as f32;
+
+                self.playback_infos[index] = Some((prev_index, k));
             } else {
-                self.internal_buffer.fill(0.);
+                self.playback_infos[index] = None;
             }
 
-            output.set_channels_values_at(index, &self.internal_buffer);
-
-            self.render_state.buffer_time += dt * computed_playback_rate;
-            self.render_state.buffer_time_elapsed += dt * computed_playback_rate;
+            let time_incr = dt * computed_playback_rate;
+            self.render_state.buffer_time += time_incr;
+            self.render_state.buffer_time_elapsed += time_incr;
             current_time += dt;
         }
 
-        true
-    }
-}
-
-impl AudioBufferSourceRenderer {
-    // Pick the closest index to the given position
-    //
-    // @note - this is not used but we keep that around as it could be usefull
-    // for testing and/or to for perf improvement if the playback is aligned on
-    // `sample_rate` and `playback_rate.abs() = 1` and as not been modified
-    #[allow(dead_code)]
-    fn compute_playback_at_position_direct(&mut self, position: f64, sample_rate: f64) {
-        let sample_index = (position * sample_rate).round() as usize;
-
-        let iterator = self.buffer.as_ref().unwrap().channels().iter().enumerate();
-        for (channel_index, channel) in iterator {
-            self.internal_buffer[channel_index] = channel.as_slice()[sample_index];
-        }
-    }
+        // fast track, no interpolation
+        if self.is_buffer_aligned {
+            let start_index; // slice start in the buffer
+            let end_index; // slice end in the buffer
+
+            match self.playback_infos[0] {
+                Some((prev_index, _k)) => {
+                    start_index = prev_index;
+                    // check if end is within the block
+                    let res = self.playback_infos.iter().position(|&el| el.is_none());
+                    end_index = match res {
+                        Some(position) => start_index + position,
+                        None => start_index + RENDER_QUANTUM_SIZE,
+                    };
+                }
+                None => {
+                    start_index = 0;
+                    end_index = 0;
```

Comment:
Are you sure `end_index` must be zero? Setting it to 128 does not fix the build by the way..